src/components/PostDetails.tsx

```diff
@@ -98,21 +62,21 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
       </div>
 
       <div className="block">
-        {!loaded && <Loader />}
+        {!loading && <Loader />}
 
-        {loaded && hasError && (
+        {loading && err && (
           <div className="notification is-danger" data-cy="CommentsError">
             Something went wrong
           </div>
         )}
 
-        {loaded && !hasError && comments.length === 0 && (
+        {loading && !err && comments.length === 0 && (
           <p className="title is-4" data-cy="NoCommentsMessage">
             No comments yet
           </p>
         )}
 
-        {loaded && !hasError && comments.length > 0 && (
+        {loading && !err && comments.length > 0 && (
```

Comment:
The condition `loading && !err && comments.length > 0` might not be correct for displaying the comments. Typically, you would want to display the comments after loading is complete, i.e., `!loading && !err && comments.length > 0`.